### PR TITLE
Mount extra volume when necessary

### LINKED ISSUE #581: workload cluster creation fails if the file passed in the kubeconfig flag is not in the working directory
**What happened**:
cluster creation fails with the following message 

```
error searching eksa VSphereMachineConfigResponse: error: stat /<full path>/some-config.kubeconfig: no such file or directory
```
even when the file exists

**What you expected to happen**:
cluster creation to succeed 

**How to reproduce it (as minimally and precisely as possible)**:
run 
```
eksctl anywhere create cluster -f some-spec.yam --kubeconfig /full path/management.kubeconfig 
```
where /full path is different from the working directory where some-spec.yaml is located 

**Anything else we need to know?**:
This happens because all commands run inside docker using the mr-tools image, and by default only the working folder is mounted as a volume. 

**Environment**:
- EKS Anywhere Release:
- EKS Distro Release:

### PR DESCRIPTION
**Pull request by mdsgabriel**
All eksa commands run inside docker using the mr-tools image.

When creating management cluster, users only have to specify a spec file,
and the cli will mount tha path that contains the spec as a volume to be
used by the mr-tools image.

With the introduction of workload clusters, users can pass an extra flag (--kubeconfig)
that points to the kubeconfig of a management cluster.
If this kubeconfig is not in the same path as the spec then it is necessary to
mount an extra volume to make it availalbe for the cli.

Fix #581

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) mentions are not allowed in commit messages.

**The list of commits with invalid commit messages**:

- [e5c386d](https://github.com/aws/eks-anywhere/commits/e5c386d966addeff4a13ce4b3cefbe6b828100b4) Mount extra volume when necessary

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/582#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
[Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) mentions are not allowed in commit messages.

**The list of commits with invalid commit messages**:

- [7678461](https://github.com/aws/eks-anywhere/commits/7678461f730b2a13c29bbe36e8c24b2e70f2dad7) Mount extra volume when necessary

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by jiayiwang7 — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/582#pullrequestreview-799091733" title="Approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/582#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
